File: fix-payments.js
const fs = require("fs");
const { execSync } = require("child_process");

const schemaPath = "prisma/schema.prisma";
let schema = fs.readFileSync(schemaPath, "utf8");

// Add relation field to School model if not already present
if (!schema.includes("campusPaymentRecords")) {
  schema = schema.replace(
    /model School \{([^}]+)\}/,
    `model School {$1
  campusPaymentRecords CampusPaymentRecord[]
}`
  );
}

// Append CampusPaymentRecord model if not already present
if (!schema.includes("model CampusPaymentRecord")) {
  schema += `
model CampusPaymentRecord {
  id            String   @id @default(cuid())
  schoolId      String
  school        School   @relation(fields: [schoolId], references: [id], onDelete: Cascade)
  orderId       String?
  paymentId     String?
  amount        Float?
  currency      String?
  status        String?
  receiptNumber String?
  studentName   String?
  purpose       String?
  createdAt     DateTime @default(now())
}
`;
}

fs.writeFileSync(schemaPath, schema, "utf8");
console.log("✅ Added CampusPaymentRecord model and relation to prisma/schema.prisma");

console.log("🔄 Pushing database schema update...");
execSync("npx prisma db push --accept-data-loss", { stdio: "inherit" });

console.log("⚡ Generating Prisma Client types...");
execSync("npx prisma generate", { stdio: "inherit" });

console.log("🏗️ Running production build verification...");
execSync("npm run build", { stdio: "inherit" });
console.log("🎉 Build completed successfully!");

File: fix_payments.py
import subprocess
import os
import re

SCHEMA_PATH = "prisma/schema.prisma"

def update_schema():
    if not os.path.exists(SCHEMA_PATH):
        print(f"❌ Error: {SCHEMA_PATH} not found.")
        return False

    with open(SCHEMA_PATH, "r") as f:
        schema = f.read()

    # Define SmartPaymentExtensionHub model with newRenewalDate included
    model_def = """model SmartPaymentExtensionHub {
  id                 String   @id @default(uuid())
  tenantId           String?
  tenant             Tenant?  @relation(fields: [tenantId], references: [id], onDelete: Cascade)
  schoolId           String?
  school             School?  @relation(fields: [schoolId], references: [id], onDelete: Cascade)
  extensionName      String?
  extensionCode      String?
  paymentGateway     String?
  invoiceId          String?
  amountPaid         String?
  extensionPeriod    String?
  previousRenewal    String?
  newRenewalDate     String?
  transactionFeePct  Float    @default(1.5)
  aiFraudDetection   Boolean  @default(true)
  status             String   @default("ACTIVE")
  createdAt          DateTime @default(now())
  updatedAt          DateTime @updatedAt
}"""

    # Remove old definition if exists
    schema = re.sub(r'model SmartPaymentExtensionHub\s*\{[^}]*\}', '', schema)
    
    # Append new model definition
    schema += "\n\n" + model_def

    # Ensure relations exist on School
    school_match = re.search(r"model School\s*\{([^}]*)\}", schema)
    if school_match:
        school_body = school_match.group(1)
        rel = "  smartPaymentExtensionHubs SmartPaymentExtensionHub[]"
        if "smartPaymentExtensionHubs" not in school_body:
            school_body = school_body.rstrip() + "\n" + rel + "\n"
            schema = schema.replace(school_match.group(0), "model School {" + school_body + "}")

    # Ensure relations exist on Tenant
    tenant_match = re.search(r"model Tenant\s*\{([^}]*)\}", schema)
    if tenant_match:
        tenant_body = tenant_match.group(1)
        rel = "  smartPaymentExtensionHubs SmartPaymentExtensionHub[]"
        if "smartPaymentExtensionHubs" not in tenant_body:
            tenant_body = tenant_body.rstrip() + "\n" + rel + "\n"
            schema = schema.replace(tenant_match.group(0), "model Tenant {" + tenant_body + "}")

    with open(SCHEMA_PATH, "w") as f:
        f.write(schema)
    
    print("✨ Successfully added newRenewalDate to SmartPaymentExtensionHub!")
    return True

if __name__ == "__main__":
    if update_schema():
        print("🧹 Clearing Next.js cache...")
        if os.path.exists(".next"):
            import shutil
            shutil.rmtree(".next")
        
        print("⚙️ Running npx prisma generate...")
        subprocess.run(["npx", "prisma", "generate"], check=True)
        
        print("🚀 Running npm run build...")
        subprocess.run(["npm", "run", "build"], check=True)
        print("🎉 Build passed successfully!")
